iceberg-rust/src/view/transaction/operation.rs

```diff
@@ -35,19 +35,60 @@ pub enum Operation {
     UpdateProperties(Vec<(String, String)>),
 }
 
+// Tries to preserve dialect order
+fn upsert_representation(
+    current_representations: &[ViewRepresentation],
+    new_representation: ViewRepresentation,
+) -> Vec<ViewRepresentation> {
```

Comment:
This function allocates a new `Vec` every time it’s called, and you call it in a loop on line 73. If `new_representations` contains `N` elements and `current_representations` contains `M`, this results in `O(N * M)` copies and allocations.

```suggestion
fn upsert_representation(
    representations: &mut Vec<ViewRepresentation>,
    new_representation: ViewRepresentation,
) -> Vec<ViewRepresentation> {
```

Reply:
True, but I don't expect the number of representations to ever exceed single digits. There's not that many different analytics systems with incompatible representations.